### PR TITLE
Get landmark Details API Code

### PR DESCRIPTION
**Pull request by VI20475016**
Below are the Steps to Execute This API and to Get Successful Response
--------------------------------------------------------------------------
Step:-1
In Legacy APIGateway Project
Under package com.wipro.APIGateway.WebService; -> GeofenceServiceV5::GetLandmarkDetails and GeofenceServiceV5::GetLandmarkDetailsForUser, Change port in variable "String urlPath" from "26070" to "27000"
Ex:- String urlPath = "http://localhost:26070/WISE/..." to String urlPath = "http://localhost:27000/WISE/..."

Step:-2
After Above Changes, Need to Make APIGateway, JCBWebAppNewScreens and Wise Applications Up and Running
-> Login With UserName as "vina398022"

Step:-3
Hit the below API from Postman to get the Response
API:- http://localhost:8088/user/machines/getlandmarks?landmarkName=LandMark 7777&vin=HAR135WJH01501510  (GET)

Note:-
------
-> Hardcoded livelinkToken value as "37aa1b15_20240522150705" and passed this livelinkToken in header such as .header("TokenId", livelinkToken) in MachineServiceImpl::getLandmarkDetails in JCB-Machines.

### CODE REVIEW

File: JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java [hunk 1]
```diff
@@ -1205,4 +1206,152 @@ public String deleteGeofenceDetails(GeofenceRequest geofenceParam, String userNa
...
+							} else {
+								geofenceResponseList.addAll(geofenceList);
+							}
+						}
+						landmarkResponse.setLandmark(landmarkName);
+						landmarkResponse.setLandmarkDetails(geofenceResponseList);
+					}
+				} else {
+					log.info("Empty Reposnse " + LandmarkStringResponse);
+					if (LandmarkStringResponse.contains("TokenId is null or Invalid")) {
```
Comment (GokulAher) [inline JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java:1295]: pl. move the message to properties file / constant file

File: JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java [hunk 2]
```diff
@@ -1205,4 +1206,152 @@ public String deleteGeofenceDetails(GeofenceRequest geofenceParam, String userNa
...
+			final WebResource getLandmarkWebResource = client.resource(getLandmarkDetailURL).queryParams(queryParams);
+			final ClientResponse getLandmarkDataResponse = getLandmarkWebResource.accept("application/json")
+					.header("Orgkey", AppServerConstants.livelinkAppServerOrgKey)
+					.header("TokenId", livelinkToken)
+					.type("application/json")
+					.get(ClientResponse.class);
+
+			if (getLandmarkDataResponse.getStatus() == HttpServletResponse.SC_OK) {
+				String LandmarkStringResponse = getLandmarkDataResponse.getEntity(String.class);
+				if (!LandmarkStringResponse.contains("No data available")
```
Comment (GokulAher) [inline JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java:1242]: pl. move the message to properties file / constant file

File: JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java [hunk 3]
```diff
@@ -1205,4 +1206,152 @@ public String deleteGeofenceDetails(GeofenceRequest geofenceParam, String userNa
...
+						&& !LandmarkStringResponse.contains("TokenId is null or Invalid")) {
```
Comment (GokulAher) [inline JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java:1243]: pl. move the message to properties file / constant file

**Review by GokulAher — APPROVED** [review]
pl. do not use hard coded messages in the code , pl. use constant file for messages that never change and properties file for the values that may cahnge

File: JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java (hunk 3 above)
Comment (VI20475016) [reply JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java:1243]: I Will Incorporate the Changes.

File: JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java (hunk 2 above)
Comment (VI20475016) [reply JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java:1242]: I Will Incorporate the Changes.

File: JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java (hunk 1 above)
Comment (VI20475016) [reply JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java:1295]: I Will Incorporate the Changes.

Comment (GokulAher) [reply JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java:1295]: Thanks!

File: JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java (hunk 3 above)
Comment (GokulAher) [reply JCB-Machines/src/main/java/com/wipro/jcb/livelink/app/machines/service/impl/MachineServiceImpl.java:1243]: Thanks!

**Pull request merged**